test/functional/test_framework/util.py

```diff
@@ -57,18 +58,44 @@ def assert_raises_message(exc, message, fun, *args, **kwds):
     else:
         raise AssertionError("No exception raised")
 
+def assert_raises_process(returncode, output, fun, *args, **kwds):
+    """Execute a process and asserts the process return code and output.
+
+    Calls function `fun` with arguments `args` and `kwds`. Catches a CalledProcessError
+    and verifies that the return code and output are as expected. Throws AssertionError if
+    no CalledProcessError was raised or if the return code and output are not as expected.
+
+    Args:
+        returncode (int), optional: the process return code. Set to None if checking the
+            return code is not required.
+        output (string), optional: [a substring of] the process output. Set to None if
+            checking the process output is not required.
+        fun (function): the function to call. This should execute a process.
+        args*: positional arguments for the function.
+        kwds**: named arguments for the function.
+    """
+    try:
+        fun(*args, **kwds)
+    except CalledProcessError as e:
+        if (returncode is not None) and (returncode != e.returncode):
```

Comment:
You can remove `(returncode is not None)` here and `(output is not None)` below. The only reason I allowed `assert_raises_jsonrpc()` callers to not specify `returncode` or `output` was to be backwards compatible with existing tests which didn't specify those. As far as I'm concerned, tests should always assert return codes and error messages if they're expecting.

(we can probably remove the same from `assert_raises_jsonrpc()` as well in a separate PR now)

Reply:
Great, will do.